### PR TITLE
fix validator set proposer priorities in light client provider

### PR DESCRIPTION
**Pull request by cmwaters**
## Description

Created `ValidatorSetFromExistingValidators` which restores a validator set (without incrementing priorities) from an array of validators.

Closes: #5306

@erikgrinaker is it possible to see if this fixes your problem

### CODE REVIEW

**Comment by erikgrinaker** [comment]
Great, thanks! I'll give it a try when I'm able.

**Comment by erikgrinaker** [comment]
This appears to give the correct light client results. However, I'm still seeing state sync consensus failures due to proposer problems, so I'm going to dig into it a bit further before approving this.

**Comment by cmwaters** [comment]
Ok. Has the output from the script you wrote here: https://github.com/tendermint/tendermint/issues/5295#issuecomment-682531663 changed at all?

**Comment by erikgrinaker** [comment]
Yes, I'm getting the right priorities, so it appears to fix the immediate problem. Not sure why I'm still seeing consensus failures.

**Review by erikgrinaker — APPROVED** [review]
There was an additional off-by-one issue elsewhere, have state sync working again now. Thanks for fixing this!

**Comment by erikgrinaker** [comment]
PS: this should have a changelog entry

File: types/validator_set.go [hunk 1]
```diff
@@ -966,6 +984,21 @@ func ValidatorSetFromProto(vp *tmproto.ValidatorSet) (*ValidatorSet, error) {
 	return vals, vals.ValidateBasic()
 }
 
+// ValidatorSetFromExistingValidators takes an existing array of validators and rebuilds
+// the exact same validator set that corresponds to it without changing the proposer priority or power
+func ValidatorSetFromExistingValidators(valz []*Validator) (*ValidatorSet, error) {
```
Comment (melekes) [inline types/validator_set.go:989]: can't we combine this func with NewValidatorSet?

Comment (cmwaters) [reply types/validator_set.go:989]: in what way? have a check to see if the proposer priority is not nil for the validators? and if so then follow the logic of this function instead

Comment (cmwaters) [reply types/validator_set.go:989]: I think my reason for having a separate function is that I wanted to show the clear distinction between the concept of making a validator set from new validators and making a validator set from existing validators. I can change this if you prefer going for simplicity

Comment (melekes) [reply types/validator_set.go:989]: > have a check to see if the proposer priority is not nil for the validators? and if so then follow the logic of this function instead

yeah

Comment (cmwaters) [reply types/validator_set.go:989]: Ok, I can make the change in another PR

**Pull request merged**